### PR TITLE
feat (experiment): #2065 screenshots in highlights experiment

### LINKED ISSUE #2065: Use screenshots for highlights with no metadata
There are times when we may show a highlight without metadata (for instance if it's a bookmark that we want to show because it's a bookmark but it does not have a suitable preview image) should we consider using a screenshot for that image rather than showing no image (ie a big block of colour)

### PR DESCRIPTION
**Pull request by rlr**
r? @jaredkerim 

fixes #2065

### CODE REVIEW

File: common/recommender/Baseline.js [hunk 1]
```diff
@@ -109,10 +109,12 @@ class Baseline {
...
-    // Boost boomarks even if they have low score or no images giving a
-    // just-bookmarked page a near-infinite boost
-    if (features.bookmarkAge) {
-      newScore += BOOKMARK_AGE_DIVIDEND / features.bookmarkAge;
+    if (this.options.experiments && this.options.experiments.bookmarkScreenshots) {
+      // Boost boomarks even if they have low score or no images giving a
+      // just-bookmarked page a near-infinite boost
+      if (features.bookmarkAge) {
+        newScore += BOOKMARK_AGE_DIVIDEND / features.bookmarkAge;
+      }
```
Comment (rlr) [inline common/recommender/Baseline.js]: @jaredkerim do we still want this change? I know there was some discussion, but not sure what was decided.

File: content-test/addon/Feeds/HighlightsFeed.test.js [hunk 2]
```diff
@@ -114,6 +125,64 @@ describe("HighlightsFeed", () => {
...
+    it("should add screenshots to sites that qualify for a screenshot if the experiment is enabled", () => {
+      reduxState.Experiments.values.bookmarkScreenshots = true;
+      instance.baselineRecommender = {scoreEntries: sinon.spy(links => links)};
+      instance.shouldGetScreenshot = () => true;
+      return instance.getData().then(result => {
+        // assert.calledTwice(instance.getScreenshot);
+        // for (let link of result.data) {
+        //   assert.equal(link.screenshot, testScreenshot);
+        //   assert.notEqual(link.metadata_source.indexOf("+Screenshot"), -1);
+        // }
```
Comment (rlr) [inline content-test/addon/Feeds/HighlightsFeed.test.js]: oops. forgot to uncomment this. fixing

File: experiments.json [hunk 3]
```diff
@@ -178,5 +178,20 @@
...
+  "bookmarkScreenshots": {
+    "name": "Screenshots for Highlights",
+    "active": true,
+    "description": "Show a screenshot for Highlights that otherwise don't have an image",
+    "control": {
+      "value": false,
+      "description": "Don't show screenshots in Highlights"
+    },
+    "variant": {
+      "id": "exp-011-bookmark-screenshots",
```
Comment (jaredkerim) [inline experiments.json]: this should be exp-013

File: addon/Feeds/HighlightsFeed.js [hunk 4]
```diff
@@ -91,6 +149,14 @@ module.exports = class HighlightsFeed extends Feed {
           this.baselineRecommender.updateOptions({highlightsCoefficients});
           this.refresh("coefficients were changed");
         }
+        if (action.data.name === "experiments.bookmarkScreenshots" && this.baselineRecommender) {
+          let highlightsCoefficients = this.getCoefficientsFromPrefs();
```
Comment (jaredkerim) [inline addon/Feeds/HighlightsFeed.js]: Why not just call:

    initializeRecommender(reason)

Comment (rlr) [inline addon/Feeds/HighlightsFeed.js]: that would've been too easy lol

**Comment by as-pine-proxy** [comment]
[Treeherder: 84ba19887b668b16c7d2e182f6d61524a2904b9e](https://treeherder.mozilla.org/#/jobs?repo=pine&revision=84ba19887b668b16c7d2e182f6d61524a2904b9e)

**Pull request merged**